File: .tmp_py4013/batch_08_dump/2863_maximum_length_of_semi_decreasing_subarrays.js
// LeetCode 2863 - Maximum Length of Semi-Decreasing Subarrays
// https://leetcode.com/problems/maximum-length-of-semi-decreasing-subarrays/

/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubarrayLength = function(nums) {
    const n = nums.length;
    let ans = 0;
    const st = [];
    for (let i = n - 1; i >= 0; i--) {
        if (!st.length || nums[i] > nums[st[st.length - 1]]) st.push(i);
    }
    for (let i = 0; i < n; i++) {
        while (st.length && nums[i] > nums[st[st.length - 1]]) {
            const j = st.pop();
            if (j - i + 1 > ans) ans = j - i + 1;
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check_12/2863_maximum_length_of_semi_decreasing_subarrays.rs
struct Solution;
// LeetCode 2863 - Maximum Length of Semi-Decreasing Subarrays
// https://leetcode.com/problems/maximum-length-of-semi-decreasing-subarrays/

impl Solution {
    pub fn max_subarray_length(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = 0i32;
        let mut st: Vec<usize> = Vec::new();
        for i in (0..n).rev() {
            if st.is_empty() || nums[i] > nums[*st.last().unwrap()] {
                st.push(i);
            }
        }
        for i in 0..n {
            while !st.is_empty() && nums[i] > nums[*st.last().unwrap()] {
                let j = st.pop().unwrap();
                ans = ans.max((j - i + 1) as i32);
            }
        }
        ans
    }
}

fn main() {}
